src/main/scala/com/mwz/sonar/scala/junit/JUnitSensor.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Sonar Scala Plugin
+ * Copyright (C) 2018 All contributors
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package junit
+
+import java.io.File
+import java.nio.file.Path
+
+import com.mwz.sonar.scala.util.PathUtils._
+import com.mwz.sonar.scala.util.{Log, MetricUtils}
+import org.sonar.api.batch.fs.{FileSystem, InputFile}
+import org.sonar.api.batch.sensor.{Sensor, SensorContext, SensorDescriptor}
+import org.sonar.api.config.Configuration
+import org.sonar.api.measures.CoreMetrics
+import org.sonar.api.scan.filesystem.PathResolver
+
+import scala.util.Try
+
+final class JUnitSensor(
+  untTestsReportParser: JUnitReportParserAPI,
+  config: Configuration,
+  fs: FileSystem,
+  pathResolver: PathResolver
+) extends Sensor {
+  import JUnitSensor._ // scalastyle:ignore org.scalastyle.scalariform.ImportGroupingChecker
+
+  private[this] val log = Log(classOf[JUnitSensor], "junit")
+
+  override def describe(descriptor: SensorDescriptor): Unit = {
+    descriptor
+      .name(SensorName)
+      .onlyOnLanguage(Scala.LanguageKey)
```

Comment:
I believe this sensor only works for test files.
Thus, you may add
`.onlyOnFileType(InputFile.Type.TEST)`

Reply:
yes definitely